### ISSUE TITLE
[feature suggestrion] Make F11 a shortcut for full-screen.

### ISSUE THREAD

**Issue by yurivict** [OpenMusicKontrollers/synthpod#14]
Now many programs have such shortcut. Uses should be able to make windows full-screen to focus on what they are doing.

**Comment by yurivict** [comment]
I noticed that Alt-F11 is full-screen shortcut. Could you please consider to make it F11 to be in like what browsers, e-mail clients and a lot of other software has - just F11?

**Comment by ventosus** [comment]
There is no shortcut to do fullscreen and the F11 key is not handled by synthpod's UI.

I won't add any shortcut for such a thing, as it is clearly the window manager's job to do that (and it'll do it consistently without any app needing to be unnecessarily be modified and agree to one arbirary key)

**Comment by yurivict** [comment]
> it is clearly the window manager's job to do that

Not all WMs support this function. This is why many programs support F11: chrome, firefox, thunderbird, gimp, meld, qTox, ...

But anyways ...

**Comment by ventosus** [comment]
> Not all WMs support this function
Really? I've never come across one

Will reopen and reconsider, then :)

**Comment by ventosus** [comment]
I've done some research and go back to my initial argument, that this is the WM's job.

In order to be able to toggle fullscreen mode, the WM needs to support the EWMH spec [1].

In a WM not implementing that spec (and the vast majority does), apps simply cannot toggle fullscreen mode by themselves, which turns your argument moot.

[1] https://standards.freedesktop.org/wm-spec/latest/

**Comment by yurivict** [comment]
> In a WM not implementing that spec (and the vast majority does), apps simply cannot toggle fullscreen mode by themselves, which turns your argument moot.

I see. So, designers of Firefox, Chrome, Opera, and other systems are simply silly people, couldn't figure out this thing.

**Comment by ventosus** [comment]
Synthpod is advertised as *lightweight*. Everything that is not strictly necessary for its proper functioning, won't be added.

If you should have a profound argument, I'm willing to engage in a fruitful discussion. Up to now you have not even managed to tell me what WM(s) do not support fullscreening by themselves...

But throughing sarcastic nonsense at me, definitely won't help, sorry.

**Comment by yurivict** [comment]
Having a meaningful discussion will be impossible since you obviously made your mind, and this also isn't a life-n-death important matter.

Sorry, that was an off-the-calf remark, half sarcastic but also true.

As for examples, there are several minimalistic window managers. I checked one, dwm. It doesn't support a shortcut for full-screen, but apps can full-screen themselves when they define such a shortcut themselves. I tried this in qTox, it worked. And dwm is only ~2,000 lines of C.

Again, this isn't a very important matter.

Also, thank you for helping to make SynthPod work on FreeBSD. I appreciate it!
